pkg/resource/deploy/deployment_executor.go

```diff
@@ -388,17 +388,6 @@ func (ex *deploymentExecutor) performDeletes(
 		logging.V(4).Infof("deploymentExecutor.Execute(...): antichain complete")
 	}
 
-	// After executing targeted deletes, we may now have resources that depend on the resource that
-	// were deleted.  Go through and clean things up accordingly for them.
-	if targetsOpt.IsConstrained() {
-		resourceToStep := make(map[*resource.State]Step)
-		for _, step := range deleteSteps {
-			resourceToStep[ex.deployment.olds[step.URN()]] = step
-		}
-
-		ex.rebuildBaseState(resourceToStep, false /*refresh*/)
-	}
```

Comment:
This is the problematic code. It's assuming all the delete steps will be successful, removing the associated resources from the state, but that may not be the case if a delete fails.

Originally, I was thinking we'd have to rework this to only do it for successful delete operations, but thinking about it more, I _think_ we don't have to call `ex.rebuildBaseState` at all for targeted deletes because if we're doing a targeted delete, we're already forced to specify dependents (or specify TargetDependents), so there shouldn't be a need to clean things up. Does this sound right, or am I missing something?

Reply:
@Frassle, @pgavlin, can you sanity check me on this?

Reply:
>  we're already forced to specify dependents

Are we? I think we check that only targetted resources or their dependents _can_ be deleted, but it looks like the planner would allow deleting a parent and not a child? It probably shouldn't but do we have a test to prove that either way?

Reply:
The PR adds a test for this: https://github.com/pulumi/pulumi/pull/14735/files#diff-aa36a122dd8ff659d8ba8df709cf3fac67dce2b07d5ad8c51240eb76b08d7022R1664-R1666

If I try to target a parent and not its child, it errors.